src/js/playlist/playlist.js

```diff
@@ -31,7 +31,7 @@ export function normalizePlaylistItem(model, item, feedData) {
 
     playlistItem.preload = getPreload(item.preload, preload);
 
-    playlistItem.allSources = formatSources(item, model);
+    playlistItem.allSources = formatSources(playlistItem, model);
```

Comment:
with the `allSources = ` on the left side, this code does not read as something that would mutate the incoming source. I would rather patch the fact that we are mutating the object within `formatSources` rather than attempting to patch this by mutating the object we eventually use elsewhere in the stack. https://github.com/jwplayer/jwplayer/blob/master/src/js/playlist/playlist.js/#L85-L86

Feels like this could just lead to problems in the future

Reply:
Are we muting it? Or are we creating a list or new sources that is filtered below?

Reply:
Ah we are mutating the sources...

Reply:
This doesn't change that, it just passed in the correct item to copy `preload` from, to the sources.